accelerator/multi_region_modules/hubNetworking.bicep

```diff
@@ -0,0 +1,1164 @@
+          serviceName: subnet.delegation
+        }
+      }
+    ]
+
+    networkSecurityGroup: (subnet.name == 'AzureBastionSubnet' && parAzBastionEnabled) ? {
+      id: '${resourceGroup().id}/providers/Microsoft.Network/networkSecurityGroups/${parAzBastionNsgName}'
+    } : (empty(subnet.networkSecurityGroupId)) ? null : {
+      id: subnet.networkSecurityGroupId
+    }
+
+    routeTable: (empty(subnet.routeTableId)) ? null : {
+      id: subnet.routeTableId
+    }
+  }
+}]
+
+var varSecondarySubnetProperties = [for subnet in varSecondarySubnetMap: {
+  name: subnet.name
+  properties: {
+    addressPrefix: subnet.ipAddressRange
+
+    delegations: (empty(subnet.delegation)) ? null : [
+      {
+        name: subnet.delegation
+        properties: {
+          serviceName: subnet.delegation
+        }
+      }
+    ]
+
+    networkSecurityGroup: (subnet.name == 'AzureBastionSubnet' && parAzBastionEnabled) ? {
+      id: '${resourceGroup().id}/providers/Microsoft.Network/networkSecurityGroups/${parAzBastionNsgName}'
+    } : (empty(subnet.networkSecurityGroupId)) ? null : {
+      id: subnet.networkSecurityGroupId
+    }
+
+    routeTable: (empty(subnet.routeTableId)) ? null : {
+      id: subnet.routeTableId
+    }
+  }
+}]
+
+var varVpnGwConfig = ((parVpnGatewayEnabled) && (!empty(parVpnGatewayConfig)) ? parVpnGatewayConfig : json('{"name": "noconfigVpn"}'))
+
+var varErGwConfig = ((parExpressRouteGatewayEnabled) && !empty(parExpressRouteGatewayConfig) ? parExpressRouteGatewayConfig : json('{"name": "noconfigEr"}'))
+
+var varGwConfig = [
+  varVpnGwConfig
+  varErGwConfig
+]
+
+// Customer Usage Attribution Id Telemetry
+var varCuaid = '2686e846-5fdc-4d4f-b533-16dcb09d6e6c'
+
+// ZTN Telemetry
+var varZtnP1CuaId = '3ab23b1e-c5c5-42d4-b163-1402384ba2db'
+var varZtnP1Trigger = (parDdosEnabled && parAzFirewallEnabled && (parAzFirewallTier == 'Premium')) ? true : false
+
+var varAzFirewallUseCustomPublicIps = length(parAzFirewallCustomPublicIps) > 0
+
+//DDos Protection plan will only be enabled if parDdosEnabled is true.
+resource resDdosProtectionPlan 'Microsoft.Network/ddosProtectionPlans@2023-02-01' = if (parDdosEnabled) {
+  name: parDdosPlanName
+  location: parLocation
+  tags: parTags
+}
+
+// Create resource lock if parDdosEnabled is true and parGlobalResourceLock.kind != 'None' or if parDdosLock.kind != 'None'
+resource resDDoSProtectionPlanLock 'Microsoft.Authorization/locks@2020-05-01' = if (parDdosEnabled && (parDdosLock.kind != 'None' || parGlobalResourceLock.kind != 'None')) {
+  scope: resDdosProtectionPlan
+  name: parDdosLock.?name ?? '${resDdosProtectionPlan.name}-lock'
+  properties: {
+    level: (parGlobalResourceLock.kind != 'None') ? parGlobalResourceLock.kind : parDdosLock.kind
+    notes: (parGlobalResourceLock.kind != 'None') ? parGlobalResourceLock.?notes : parDdosLock.?notes
+  }
+}
+
+resource resHubVnets 'Microsoft.Network/virtualNetworks@2023-02-01' = [for i in range(0, length(varLocations)): {
```

Comment:
If we will deploy just one region using this module then we will need to drop the loop deployments